lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -570,4 +573,108 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    @Override
+    public StatusEvent updateEvent(
+            StatusEvent ev)
+            throws IndexingException,
+                   RetriableIndexingException {
+
+        // latest states to update
+        StatusEventType latestEventType = ev.getEventType();
+        Boolean latestIsPublic = null;
+        String latestName = null;
+
+        if( ev.isPublic().isPresent()) {
+            latestIsPublic = ev.isPublic().get().booleanValue();
+        }
+        if( ev.getNewName().isPresent()) {
+            latestName = ev.getNewName().get().toString();
+        }
+
+
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        long wsid = Long.valueOf(ev.getAccessGroupId().get().toString()).longValue();
+        long objid = Long.valueOf(ev.getAccessGroupObjectId().get().toString()).longValue();
+
+        Map<String, Object> command;
+
+        // check if object is deleted and update state of latestEventType
```

Comment:
Why not just make the get object info call below, and if it throws a JsonClientException containing a string indicating the workspace or object was deleted, mutate the event to DELETE_ALL_VERSIONS?

Reply:
I thought about this approach but felt too hacky to depend on the an exception message substring to make a decision.

Reply:
Personally I'd like to update the sdk to add error codes to exceptions like the auth service. That being said, there will be times where we need to discriminate between exceptions and right now the only way to do that is via the string. We already do it in this class, so I'm ok with it. My personal opinion is that the simpler code is worth the hackiness but I leave the call up to you.